clcache.py

```diff
@@ -737,6 +737,21 @@ def expandCommandLine(cmdline):
 class CommandLineAnalyzer(object):
 
     @staticmethod
+    def _preprocessToStdout(options):
+        assert ('P' in options or 'E' in options or 'EP' in options), "must be in preprocessor mode"
+
+        # Note: For MSVS 2013 and 2015 there is a documentation bug in
+        # https://msdn.microsoft.com/en-us/library/becb7sys.aspx stating that
+        # "To send the preprocessed output to stdout, with #line directives,
+        # use /P and /EP together."
+        # The actual compiler behavior is that both /P and /P /EP go to file.
+        # /P is with #line annotations and /P /EP is without.
+        if 'P' in options:
+            return False
+        else:
+            return True
```

Comment:
Is the `assert` maybe a bit harsh? I think Python permits expressing whether some options cause preprocessor output to be printed to `stdout` in almost plain english:

```python
def _preprocessToStdout(options):
    return ('E' in options or 'EP' in options) and 'P' not in options
```

I think that reads very nicely.